TACC/tup-ui
# Task/TUP-457/TUP-514 Add System Queue Details #234 - support smaller windows

## Overview

Make new systems section accessible on small windows.

## Related

- tweaks #234

## Changes

- **removed** old `SystemDetails` CSS*
- **added** new `SystemDetails` CSS*†
- **changed** `SectionDetails` markup to:
    - wrap only the table that needs to scroll
    - make able to scroll the table that should scroll
    - use markup similar to Dashboard†

<sup>* The `SystemNavBar` CSS is untouched. It should be moved out to its own CSS file; I'll comment about this in #234.</sup>
<sup>† The `Dashboard` layout is most similar to the new design. I referenced its markup and CSS.</sup>

## Testing

1. Open http://localhost:8000/portal/system_status 
2. View layout on wide screen:
    1. All panels with content should stack vertically.
    2. _The panel without content should be on the fight side of those with content._
    3. All panels should have the same space between them.
3. View layout on narrow screen:
    1. All panels (with and without content) should stack vertically.
    2. _The panel without content should be on the bottom._
    3. All panels should have the same space between them.

## UI

| | inner grid | outer gird |
| - | - | - |
| narrow | ![narrow inner](https://github.com/TACC/tup-ui/assets/62723358/de2ebdca-41d3-4cd2-95a0-e40af9b027a1) | ![narrow outer](https://github.com/TACC/tup-ui/assets/62723358/2c928a89-8010-4c2b-8f45-cb06504f7cf3) |
| almost narrow | ![boundary inner](https://github.com/TACC/tup-ui/assets/62723358/a78e756e-f619-4487-a4fe-5f422dfda2d6) | ![boundary outer](https://github.com/TACC/tup-ui/assets/62723358/2e7f2596-4a54-4530-a611-a2ecf66c8f98) |
| wide | ![wide inner](https://github.com/TACC/tup-ui/assets/62723358/fe605c4e-6b7f-4055-b8bf-0c7936d4c36a) | ![wide outer](https://github.com/TACC/tup-ui/assets/62723358/23fcfd80-f319-4396-8497-12e56439827c) |

## Notes

1. This grid layout is too complex. If I simplify it now, the similarity with Dashboard grid is difficult to see and thus learn from. But if I don't simplify it, I set a precedent of copying overly complex grids. I'll probably simplify it…
2. The narrow screen shows extra space instead of "Avg. Wait Time" placeholder content.
2. The "Avg. Wait Time" should not be committed.
